### PR TITLE
Adapt Tools CLI to uv

### PR DESCRIPTION
**Pull request by vinibrsl**
This pull request adapts commands from the Tool CLI from Poetry to uv.

Extra comments inline 👇

### CODE REVIEW

File: src/crewai/cli/plus_api.py [hunk 1]
```diff
@@ -25,7 +25,9 @@ def __init__(self, api_key: str) -> None:
 
     def _make_request(self, method: str, endpoint: str, **kwargs) -> requests.Response:
         url = urljoin(self.base_url, endpoint)
-        return requests.request(method, url, headers=self.headers, **kwargs)
+        session = requests.Session()
+        session.trust_env = False
```
Comment (vinibrsl) [inline src/crewai/cli/plus_api.py:29]: `trust_env = False` tells `requests` to bypass the system's authentication or proxy settings, relying only on the authentication defined in the request itself.

The reason for this change is that when a `.netrc` file is present, `requests` automatically replaces the `Authorization` header with the credentials from `.netrc`, overwriting Auth0 credentials.

While `trust_env = False` fixes the issue, alongside I'm proposing moving our custom PyPI index endpoints from `app.crewai.com` to `tools.crewai.com`. This ensures `.netrc` is only applied to that specific subdomain system-wide.

File: src/crewai/cli/tools/main.py [hunk 2]
```diff
@@ -159,23 +161,19 @@ def login(self):
             "Successfully authenticated to the tool repository.", style="bold green"
         )
 
-    def _set_netrc_credentials(self, credentials):
-        # Create .netrc or _netrc file
-        netrc_filename = "_netrc" if platform.system() == "Windows" else ".netrc"
-        netrc_path = Path.home() / netrc_filename
+    def _set_netrc_credentials(self, credentials, netrc_path=None):
```
Comment (vinibrsl) [inline src/crewai/cli/tools/main.py:164]: I refactored it to use Python's builtin `netrc` package.

File: src/crewai/cli/tools/main.py [hunk 3]
```diff
@@ -159,23 +161,19 @@ def login(self):
...
+            netrc_filename = "_netrc" if platform.system() == "Windows" else ".netrc"
+            netrc_path = Path.home() / netrc_filename
+            netrc_path.touch(mode=stat.S_IRUSR | stat.S_IWUSR, exist_ok=True)
 
-        netrc_content = f"""machine app.crewai.com
-login {credentials['username']}
-password {credentials['password']}
-"""
+        netrc_instance = netrc(file=netrc_path)
+        netrc_instance.hosts["app.crewai.com"] = (credentials["username"], "", credentials["password"])
```
Comment (vinibrsl) [inline src/crewai/cli/tools/main.py:171]: This is the line we need to change after we migrate to `tools.crewai.com`.

File: src/crewai/cli/plus_api.py (hunk 1 above)
Comment (gvieira) [reply src/crewai/cli/plus_api.py:29]: Your suggestion makes total sense.

**Review by gvieira — APPROVED** [review]

**Pull request merged**